tools/metadata_loader/main.py

```diff
@@ -0,0 +1,171 @@
+import os
+import json
+import requests
+import argparse
+from dotenv import load_dotenv
+from azure.storage.blob import BlobServiceClient, ContainerClient, BlobClient
+from urllib.parse import quote
+
+# given the datasource details
+# given the folder
+# enumerate the files in the folder
+# for each one
+#     read the file
+#     call the create meta api
+
+def get_config():
+    load_dotenv()
+    return {
+        "API_URL_BASE": os.environ.get("API_URL_BASE"),
+        "STORAGE_ACCOUNT_URL": os.environ.get("STORAGE_ACCOUNT_URL"),
+        "STORAGE_CONNECTION_STRING": os.environ.get("STORAGE_CONNECTION_STRING"),
+        "STORAGE_SAS_KEY": os.environ.get("STORAGE_SAS_KEY")
+    }
+
+def call_get_datasources_api(url):
+    return requests.get(url)
+
+def get_datasources(args):
+
+    config = get_config()
+
+    url = f'{config["API_URL_BASE"]}/api/datasources'
+    resp = call_get_datasources_api(url)
+
+    print(resp.text)
+
+    return resp.text
+
+
+def call_create_datasource_api(url, payload):
+    return requests.post(url, json=payload)
+
+def create_datasource(args):
+
+    config = get_config()
+
+    url = f'{config["API_URL_BASE"]}/api/datasources'
+    data = {
+        "name": f'{args.name}',
+        "accountName": f'{args.accountName}',
+        "containerName": f'{args.containerName}',
+        "description": f'{args.description}'
+    }
+    resp = call_create_datasource_api(url, payload=data)
+
+    print(resp.text)
+
+    return resp.text
+
+
+def call_get_all_metadata_api(url:str):
+    return requests.get(url)
+    
+def get_all_meta(args):
+    
+    config = get_config()
+    
+    url = f'{config["API_URL_BASE"]}/api/datasources/{args.accountName}/{args.containerName}/meta'
+    resp = call_get_all_metadata_api(url)
+
+    items = json.loads(resp.text)
+    if len(items) == 0:
+        print(f"There are no items in account: {args.accountName}, container: {args.containerName}.")
+    else:
+        for item in items:
+            print(f"Account: {item['accountName']}, Container: {item['containerName']}, filepath: {item['filepath'].replace('(slash)', '/')}")
+
+    return resp.text
+
+def call_create_meta_api(url, payload):
+    return requests.post(url, json=payload)
+
+def create_meta(accountName: str, containerName: str, filepath: str, document: str):
+
+    config = get_config()
+
+    #quoted_filepath = quote(filepath, safe='')
+    quoted_filepath = filepath.replace("/", "(slash)")
```

Comment:
We will be using url encoding instead of just (slasj) to pass filenames there are plenty of other charss that are not url safe but that are encoded auttomatically by the url enconding function.

Reply:
I tried quoting the string, but the api converted the %2F back to /. So in the interim, I changed it to (slash).